chore(config): add authentication table (#28)
flipt-io/docs: operations/deployment.mdx

---
title: Deployment
description: Details on various ways to deploy Flipt.
---

Flipt is built and delivered as a single standalone Linux binary.
Head to the [Installation](/installation) section for more details.

Running Flipt with the default configuration is a great way to get to grips with using it.

> However, the default settings have limitations, which might make it impractical in a production environment.

Reviewer: ```suggestion
However, the default settings have some limitations which might make it impractical in a production environment.
```

This guide will explore some Flipt deployment configurations for increased scalability and reliability.

## The Defaults

> By default, Flipt can only run as a single instance process backed by SQLite. Additionally, all API interactions go directly to this database.

Reviewer: ```suggestion
By default, Flipt runs as a single instance process backed by SQLite. Additionally, all API interactions go directly to this database.
```

This has the potential for multiple benefits:

- Failures in either read or write tiers can be isolated from one another. A failure in the tier serving the dashboard wouldn’t necessarily affect flag evaluations.
- Reads could be configured with access to the cache, where writes could be isolated from the caching tier.  This may have benefits to the number of connections required on your caching layer.
- Reads could be deployed in front of read-only replicas of a database, with the write-tier connecting directly to the primary, allowing more potential for scale in the database layer.

<Frame>
<div className="block dark:hidden">
  ![Flipt separate read/write tier deployment configuration](images/operations/flipt-complex-deployment-light.svg)
</div>

<div className="hidden dark:block">
  ![Flipt separate read/write tier deployment configuration](images/operations/flipt-complex-deployment-dark.svg)
</div>
</Frame>

> Flipt ships with metrics, logging and tracing around both behaviour and system performance metrics. These pieces of telemetry can be useful for understanding the constraints within your setup of Flipt. We recommend reading the observability section of the configuration documentation to understand more about how to extract these measurements.

Reviewer: should we link to the observability docs here?

Author: Good shout
